### ISSUE TITLE
ExpoErrorRecovery breaks on strict privacy settings

### ISSUE THREAD

**Issue by SleeplessByte** [expo/expo#8355]
### Steps to Reproduce

- Create a new expo project
- Build for web
- Deploy somewhere
- iFrame the deployed content (without adding CSP/CORS headers)
- Open in Firefox with Privacy on custom > block cookies from unvisited websites

### Expected Behavior

It should complain about the serviceworker, and no data should be persistable to localstorage, but everything else works.

### Actual Behavior

`ExpoErrorRecovery` tries to read and write from and to localstorage, resulting into a SecurityException and the entire app breaks.

### Solution

The `canUseDom` flags don't capture this:

https://github.com/expo/expo/blob/52a1e3b61343c7cf9589ec7288e5761d0137e606/packages/expo-error-recovery/src/ExpoErrorRecovery.web.ts#L5

Wrap the function in a `try {} catch`.

**Comment by SleeplessByte** [comment]
```
⚠ Request to access cookie or storage on “<URL>” was blocked because we are blocking all third-party storage access requests and content blocking is enabled. 2

❗ SecurityError: The operation is insecure. 2.2294dee0.chunk.js:1
ℹ Failed to register service-worker DOMException: "The operation is insecure." register-service-worker.js:11:16
❗ Failed to register/update a ServiceWorker for scope ‘https://branding-preview.introcloud.app/’: Storage access is restricted in this context due to user settings or private browsing mode.
```

**Comment by SleeplessByte** [comment]
```
⚠ Request to access cookie or storage on “<URL>” was blocked because we are blocking all third-party storage access requests and content blocking is enabled. 2

❗ SecurityError: The operation is insecure. 2.2294dee0.chunk.js:1
ℹ Failed to register service-worker DOMException: "The operation is insecure." register-service-worker.js:11:16
❗ Failed to register/update a ServiceWorker for scope ‘https://something.app/’: Storage access is restricted in this context due to user settings or private browsing mode.
```

**Comment by LinusU** [comment]
@tommybru @cruzach would it be possible to publish a new version of `expo-error-recovery` to Npm with this fix in it? We are currently seeing this in a public app, and would prefer to fix it by bumping the package ☺️

**Comment by brentvatne** [comment]
@LinusU - i published expo-error-recovery@1.2.2 including @tommybru's fix. can i get eyes on and possibly a merge/publish for https://github.com/LinusU/react-native-get-random-values/pull/13 in return? 😄

**Comment by LinusU** [comment]
Thanks! Hehe, of course, sorry for the delay!
